tfjs-layers/src/layers/nlp/tokenizers.ts

```diff
@@ -0,0 +1,134 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+/**
+ *  Tokenizer layers.
+ */
+
+/* Original source: keras-nlp/tokenizer.py */
+import { Tensor1D, serialization, tensor1d } from '@tensorflow/tfjs-core';
+
+import { Layer } from '../../engine/topology';
+import { NotImplementedError, ValueError } from '../../errors';
+
+export declare interface TokenizerOptions {
+  mode?: 'tokenize' | 'detokenize'
+}
+
+/**
+ * Base class for Tokenizers.
+ *
+ * Subclassers should always implement the `tokenize()` method, which will also
+ * be the default when calling the layer directly on inputs.
+ */
+export abstract class Tokenizer extends Layer {
```

Comment:
Let's add the rest of the Keras documentation here (as long as it's applicable). Thanks!

Usually we put the docs in exports_layers.ts, but that's not possible in this PR since `Tokenizer` is an abstract class (so we can't create a function that returns an instance of it). For future layers, we should add the corresponding function to `exports_layers.ts`:

[Example](https://github.com/tensorflow/tfjs/blob/tfjs-v4.7.0/tfjs-layers/src/exports_layers.ts#L370-L403), and on the [API site](https://js.tensorflow.org/api/latest/#layers.cropping2D).

We sometimes also include runnable examples for layers, which appear in the code boxes on the API website. That would have been a good place for the WhitespaceTokenizer example that Keras has in their docs, but since this is an abstract class, I think we should put that example here instead, just so people have a reference of how to implement their own tokenizer if they want to.

Reply:
done